lib/Driver/Driver.cpp

```diff
@@ -1301,6 +1319,20 @@ void Driver::buildActions(const ToolChain &TC,
                                              previousBuildState));
           AllModuleInputs.push_back(Current.get());
         }
+        if (PCH) {
+          // When we have a PCH job, it's officially owned by the Actions array;
+          // but it's also a secondary input to each of the current JobActions,
+          // which means that we need to flip the "owns inputs" bit on the
+          // JobActions so they don't try to free it. That in turn means we need
+          // to transfer ownership of all the JobActions' existing inputs to the
+          // Actions array, since the JobActions either own or don't own _all_
+          // of their inputs. Ownership can't vary input-by-input.
```

Comment:
Can you tag this with a FIXME? I want to go back and clear up the Action ownership muck at some point.